src/routing/core.js

```diff
@@ -52,7 +54,10 @@ export function setup(app) {
   app.use([coreBuildRoutes, "/narratives/*"], setSource(req => new CoreSource())); // eslint-disable-line no-unused-vars
 
   app.routeAsync(coreBuildRoutes)
-    .all(setDataset(req => req.path), canonicalizeDataset(path => `/${path}`))
+    .all(
+      setDataset(req => req.path),
+      canonicalizeDataset((req, path) => `/${path}` + (Object.keys(req.query).length ? `?${querystring.stringify(req.query)}` : ''))
```

Comment:
_non-blocking_

I think something like the following reads cleaner:

```suggestion
      canonicalizeDataset((req, path) => url.format({pathname: `/${path}`, query: req.query}))
```

and it should have the same behaviour.

(Ideally we'd be using the modern `URL` and `URLSearchParams` interfaces like we do elsewhere, but those are a bit more verbose in this situation without helper funcs. Also, ideally we'd be preserving the exact query string as-is without roundtripping thru a deserialize/serialize transform.)

Reply:
Switched to `url.format` which nicely matches our other usage across the codebase.